### PR TITLE
[web] Implement ErrorPage

### PR DESCRIPTION
**Pull request by asiia-trilitech**
## Proposed changes

[Task link](https://github.com/trilitech/umami-security-review/issues/40)

Added for web, slightly updated for desktop

## Types of changes

- [ ] Bugfix
- [x] New feature
- [ ] Refactor
- [ ] Breaking change
- [ ] UI fix

#### Web:
Screenshot 2024-11-26 at 16 53 17
Screenshot 2024-11-26 at 17 04 03

#### Desktop:
Screenshot 2024-11-26 at 16 40 57

## Checklist

- [x] Tests that prove my fix is effective or that my feature works have been added
- [ ] Documentation has been added (if appropriate)
- [ ] Screenshots are added (if any UI changes have been made)
- [ ] All TODOs have a corresponding task created (and the link is attached to it)

### CODE REVIEW

File: apps/web/src/components/ErrorPage/ErrorPage.test.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+describe("ErrorPage", () => {
+  it("renders the error message and buttons", () => {
+    render(<ErrorPage />);
+
+    expect(screen.getByText("Oops! Something went wrong!")).toBeInTheDocument();
+    expect(
+      screen.getByText("Please refresh the page or use one of the following options:")
+    ).toBeInTheDocument();
+    // Action buttons
+    expect(screen.getByRole("button", { name: /Report Error/i })).toBeInTheDocument();
```
Comment (OKendigelyan) [inline apps/web/src/components/ErrorPage/ErrorPage.test.tsx:16]: Please use exact matching by string, not regex

File: apps/web/src/components/ErrorPage/ErrorPage.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+              textAlign="center"
+              size="sm"
+            >
+              Please refresh the page or use one of the following options:
+            </Text>
+          </Center>
+
+          <VStack width="100%" spacing="16px">
+            <Button width="100%" onClick={saveBackup} size="lg" variant="primary">
+              Save Backup
```
[the remark this answers is not in the document]
Comment (asiia-trilitech) [reply apps/web/src/components/ErrorPage/ErrorPage.tsx]: For Onboarding flow I removed options to log out and save backup for both web & desktop

Let me check verification logic

[the remark this answers is not in the document]
Comment (asiia-trilitech) [reply apps/web/src/components/ErrorPage/ErrorPage.tsx]: If account is not verified, download backup action suggests to set password before saving the backup

File: apps/web/src/main.tsx [hunk 3]
```diff
@@ -8,12 +8,12 @@ import { BrowserRouter } from "react-router-dom";
 import { PersistGate } from "redux-persist/integration/react";
 
 import { App } from "./components/App";
+import { ErrorPage } from "./components/ErrorPage/ErrorPage";
```
Comment (OKendigelyan) [inline apps/web/src/main.tsx:11]: we can import directly from .`/components/ErrorPage` since we export ErrorPage.tsx from ErrorPage/index.js

**Review by OKendigelyan — APPROVED** [review]
LGTM with one comment

File: apps/web/src/main.tsx (hunk 3 above)
Comment (asiia-trilitech) [reply apps/web/src/main.tsx:11]: We should update lint rules with something like this

```{
  "plugins": ["import"],
  "rules": {
    "import/no-useless-path-segments": ["error", { "commonjs": true }]
  }
}
```

**Pull request merged**